3.0-RELEASE-NOTES.md

```diff
@@ -27,3 +27,67 @@ To disable a static remote method
 
 To disable a prototype remote method:
 `disableMethodByName('prototype.updateAttributes')`
+
+## New error-handler for rest-adapter
+
+The REST adapter (typically created via `loopback.rest()`) uses a new error
+handler implementation that provides more secure configuration out of the box.
+
+The environment setting `NODE_ENV='production'` is no longer supported,
+production vs. debug mode is controlled exclusively by configuration.
+Production environment is assumed by default, the insecure debug mode
+must be explicitely turned on.
+
+You can learn more about the rationale behind the new handler in
+[this comment](https://github.com/strongloop/loopback/issues/1650#issuecomment-161920555)
+
+User can specific options in their applications in `config.json` as follow:
+```json
+{
+  "restApiRoot": "/api",
+  "host": "0.0.0.0",
+  "port": 3000,
+  "errorHandler": {
+    "debug": false,
+    "log": false
+  }
+}
+```
+
+#### Production mode
+
+Stack trace is never returned in HTTP responses.
+
+Bad Request errors (4xx) provide the following properties copied from the
+error object: `name`, `message`, `statusCode` and `details`.
+
+All other errors (including non-Error values like strings or arrays) provide
+only basic information: `statusCode` and `message` set to status name from HTTP
+specification.
+
+#### Debug mode
+
+When in debug mode, HTTP responses include all properties provided by the error.
+
+For errors that are not an object, their string value is returned in
+`message` field.
+
+When a method fails with an array of errors (e.g. bulk create), HTTP the response
+contains still a single wrapper error with `details` set to the original array
+of errors.
+
+An example of an error response when an array of errors was raised:
+
+```js
+{
+  error: {
+    statusCode: 500,
+    name: 'ArrayOfErrors',
+    message: 'Failed with multiple errors, see `details` for more information.',
+    details: [
+      { name: 'Error1', message: 'expected error', statusCode: 500, stack: '<stacktrace>' },
+      { name: 'Error2', message: 'expected error2', statusCode: 500, stack: '<stacktrace>'}
+    ]
+  }
+}
+```
```

Comment:
Please add a link to this pull request so that people can find the discussion that led to this change (see other release note entries) and also a pointer to strong-error-handler so that people can find more information about the settings available in this new error handler.